hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/tools/hdfs-chmod/hdfs-chmod.cc

```diff
@@ -140,8 +140,17 @@ bool Chmod::HandlePath(const std::string &permissions, const bool recursive,
   /*
    * strtol is reading the value with base 8, NULL because we are reading in
    * just one value.
+   *
+   * The strtol function may result in errors so check for that before typecasting.
    */
-  auto perm = static_cast<uint16_t>(strtol(permissions.c_str(), nullptr, 8));
+  errno = 0;
+  long result = strtol(permissions.c_str(), nullptr, 8);
+  /*
+   * The errno is set to ERANGE incase the string doesn't fit in long
+   * Also, the result is set to 0, in case conversion is not possible
+   */
+  if ((errno == ERANGE) || result == 0) return false;
```

Comment:
@rishabh1704,
This would yield `false` even for those cases where `permissions` passed was 0. Please explore how this can be handled.

Reply:
Sure, i will handle that case as well.

Reply:
One solution that I thought of was to use [std::all_of](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of) to check if `permissions` contains only 0 and then bypass the `result == 0` if that happens to be the case.